### PR TITLE
Fix database model and fetching data from database

### LINKED ISSUE #89: When fetching fetching tweets from database, fetch tweets Author too and embed it inside database model object
This way our db tweet to api tweet converter will not need to access database and it will be able to move it out of the service class.

The problem is that im not 100% sure if this is a good design to put this kind of logic into database access objects.

### LINKED ISSUE #97: Update database model
Add another model object that stores just public data of users. It will be used most of the time.
The user struct that stores all user data will be only used for account related tasks, like logging in or registering a user.

Add another model object that stores Tweet with Author data embedded.

This is a part of #89

### PR DESCRIPTION
**Pull request by Prytu**
fixes #97 
fixes #89

### CODE REVIEW

File: backend/database/model/user.go [hunk 1]
```diff
@@ -19,3 +19,11 @@ type User struct {
 	Name          string
 	AvatarUrl     sql.NullString
 }
+
+type PublicUser struct {
+	ID        int64
+	Username  string
+	LastLogin time.Time
```
Comment (VirrageS) [inline backend/database/model/user.go]: This is unnecessary.

File: backend/database/user.go [hunk 2]
```diff
@@ -88,7 +117,20 @@ func (db *UserDB) getUserUsingQuery(query string, args ...interface{}) (*model.U
...
+}
+
+func (db *UserDB) getPublicUserUsingQuery(query string, args ...interface{}) (*model.PublicUser, error) {
+	var user model.PublicUser
+
+	row := db.QueryRow(query, args...)
+	err := row.Scan(&user.ID, &user.Username, &user.LastLogin, &user.Name, &user.AvatarUrl)
+
+	if err != nil && err != sql.ErrNoRows {
+		log.WithField("query", query).WithError(err).Error("getPublicUserUsingQuery database error.")
```
Comment (VirrageS) [inline backend/database/user.go]: When error occurs is `&user == nil` here? Even though it is I would opt for returning after this `return nil, err`.

Comment (Prytu) [inline backend/database/user.go]: yeah, I must have missed this one

**Comment by Prytu** [comment]
Fixed

**Pull request merged**